manifests/init.pp

```diff
@@ -1,135 +1,136 @@
+#
+# @param always_query_group_plugin
+#   Sets Defaults option 'always_query_group_plugin'. Previously all unknown system
+#   groups was automatically passed to the group plugin. This is no longer the case
+#   since 1.8.15. To pass unknown system groups to group_plugin 'always_query_group_plugin'
+#   must be set.
+#
+#   Sudo lines with the syntax below will always use group_plugin to resolve groups.
+#   plugin for that specific entry:
+#   <pre>
+#   %:Group
+#   </pre>
+#
+#   This option is automatically enabled if include_libsudo_vas is set to true and
+#   $::sudo_version => 1.8.15.
+#
+# @param hiera_merge_sudoers
+#   TODO: missing documentation
 #
 class sudo (
-  $package                   = 'sudo',
-  $package_source            = undef,
-  $package_ensure            = 'present',
-  $package_manage            = true,
-  $package_adminfile         = undef,
-  $config_dir                = '/etc/sudoers.d',
-  $config_dir_group          = 'root',
-  $config_dir_mode           = '0750',
-  $config_dir_ensure         = 'directory',
-  $config_dir_purge          = true,
-  $sudoers                   = undef,
-  $sudoers_manage            = true,
-  $config_file               = '/etc/sudoers',
-  $config_file_group         = 'root',
-  $config_file_owner         = 'root',
-  $config_file_mode          = '0440',
-  $requiretty                = true,
-  $visiblepw                 = false,
-  $always_set_home           = true,
-  $envreset                  = true,
-  $envkeep                   = ['COLORS','DISPLAY','HOSTNAME','HISTSIZE','INPUTRC','KDEDIR','LS_COLORS','MAIL','PS1','PS2','QTDIR','USERNAME','LANG','LC_ADDRESS','LC_CTYPE','LC_COLLATE','LC_IDENTIFICATION','LC_MEASUREMENT','LC_MESSAGES','LC_MONETARY','LC_NAME','LC_NUMERIC','LC_PAPER','LC_TELEPHONE','LC_TIME','LC_ALL','LANGUAGE','LINGUAS','_XKB_CHARSET','XAUTHORITY'], #lint:ignore:140chars
-  $secure_path               = '/sbin:/bin:/usr/sbin:/usr/bin',
-  $root_allow_all            = true,
-  $includedir                = true,
-  $include_libsudo_vas       = false,
-  $libsudo_vas_location      = 'USE_DEFAULTS',
-  $always_query_group_plugin = 'USE_DEFAULTS',
-  $hiera_merge_sudoers       = false,
+  String[1]                                      $package                   = 'sudo',
+  Optional[String[1]]                            $package_source            = undef,
+  Enum['present','absent']                       $package_ensure            = 'present',
+  Boolean                                        $package_manage            = true,
+  Optional[Stdlib::Absolutepath]                 $package_adminfile         = undef,
+  Stdlib::Absolutepath                           $config_dir                = '/etc/sudoers.d',
+  String[1]                                      $config_dir_group          = 'root',
+  Stdlib::Filemode                               $config_dir_mode           = '0750',
+  Enum['present', 'absent', 'directory', 'link'] $config_dir_ensure         = 'directory',
```

Comment:
We do not allow setting the file resource's target. Maybe there is no point in allowing `link`?

Reply:
I'll remove 'link' then.

Reply:
removed with commit [4079f70](https://github.com/Ericsson/puppet-module-sudo/pull/39/commits/4079f703a4e6c59e4a343294007acb9acdd3d313)